### PR TITLE
if `onlyTestTsNext` is set to true only install the "next" typescript

### PR DESCRIPTION
**Pull request by joshstevens19**
At the moment if you supply `--onlyTestTsNext` it still installs **all** the typescript packages when it only needs to install the `next` one. 

This fixes this.

Can we merge this in ASAP please 👍

### CODE REVIEW

**Comment by joshstevens19** [comment]
Can you please look at this @sandersn It's making our build time a lot longer then it needs to be due to installing all the ts packages.

Thanks

File: src/index.ts [hunk 1]
```diff
@@ -57,7 +57,13 @@ async function main(): Promise<void> {
 		// Do this *after* to ensure messages sent during installation aren't dropped.
 		await installAll();
 	} else {
-		await installAll();
+		if (!onlyTestTsNext) {
```
Comment (sandersn) [inline src/index.ts]: Can you invert this if please?

File: src/index.ts [hunk 2]
```diff
@@ -57,7 +57,13 @@ async function main(): Promise<void> {
...
+			await installAll();
+		} else {
+			// if `onlyTestTsNext` is true we only need to install the next typescript version
```
Comment (sandersn) [inline src/index.ts]: Remove this comment. The names onlyTestTsNext and installNext convey the intent by themselves.

**Comment by sandersn** [comment]
What does dtslint do as part of your build? It seems like overkill unless you are building definitely typed.

**Comment by joshstevens19** [comment]
Sorted them out thanks.

In `ethereum/web3` we have created all type support for our all our packages. We have chose to put this within our lib like `moment` and `redux` so we can manage it ourselves. We can then bring in contribution guidelines with maintaining types etc things we could not control much if it was outside the lib. This solves a lot of issues for our TS community but we wanted some tests written out for them just to be sure they match up, hence we used `dtslint` for this 👍 

Cheers

**Comment by joshstevens19** [comment]
@sandersn When are you planning a new NPM deployment of this?

**Comment by sandersn** [comment]
It's deployed! Thanks for the reminder -- Definitely Typed just pulls the package from the production branch on github, so npm deployment isn't a required step for me.

**Pull request merged**